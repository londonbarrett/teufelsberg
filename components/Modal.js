import * as React from 'react';
import PropTypes from 'prop-types';
import CloseButton from '../components/CloseButton';

const Modal = ({ title, children }) => (
  <section className="container">
    <div className="header">
      <span className="title">{title}</span>
      <CloseButton className="close" />
    </div>
    <div className="content">
      { children }
    </div>
    <style jsx>
      {`
        .container {
          align-items: center;
          background: black;
          display: flex;
          height: 100vh;
          position: fixed;
          top: 0;
          width: 100vw;
        }
        .header {
          position: fixed;
          top: 0;
          width: 100%;
        }
        .content {
          width: 100%;
        }
        .title {
          color: white;
          display: block;
          font-size: 2rem;
          font-weight: bold;
          line-height: 3rem;
          margin: 1rem;
          width: 80%;
        }
        .close {
          position: absolute;
        }
      `}
    </style>
  </section>
);

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
};

export default Modal;
