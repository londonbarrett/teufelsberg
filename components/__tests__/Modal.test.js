/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../Modal';

describe('Modal Tests', () => {
  it('Renders correctly', () => {
    const component = renderer.create((
      <Modal title="Modal Title">
        <p>Modal content</p>
      </Modal>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
