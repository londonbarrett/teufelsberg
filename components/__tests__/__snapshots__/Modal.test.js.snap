// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Tests Renders correctly 1`] = `
<section
  className="jsx-1346030082 container"
>
  <div
    className="jsx-1346030082 header"
  >
    <span
      className="jsx-1346030082 title"
    >
      Modal Title
    </span>
    <a
      className="jsx-348398248 link"
      href="/"
      onClick={[Function]}
    >
      X
    </a>
  </div>
  <div
    className="jsx-1346030082 content"
  >
    <p>
      Modal content
    </p>
  </div>
</section>
`;
